quinn-proto/src/connection/mod.rs

```diff
@@ -634,14 +632,16 @@ where
                 pad_datagram |= sent.requires_padding;
             }
 
+            pad_datagram |= space_id == SpaceId::Initial && self.side.is_client();
```

Comment:
Oh, this is definitely good too. But which case had been missed before that?  All non close frames should already have been handled by setting `requires-padding` in `populate_packet`

Reply:
I think close packets were the only incorrect case.

Reply:
For which case? The close case should have been handled by the `pad_datagram = true` in line 616 which is removed here. However I'm wondering if some other case is missing and fixed by this, since the unit-tests still don't work with GSO support, even after #1030 is added.

Reply:
Setting `pad_datagram = true` unconditionally above caused *all* close packets to be padded, which is incorrect.